pkg/kapp/clusterapply/add_or_update_change.go

```diff
@@ -218,6 +223,28 @@ func (c AddOrUpdateChange) recordAppliedResource(savedRes ctlres.Resource) error
 		return fmt.Errorf("Calculating change after the save: %s", err)
 	}
 
+	// Use compact representation to take as little space as possible
+	// because annotation value max length is 262144 characters
+	// (https://github.com/vmware-tanzu/carvel-kapp/issues/48).
+	appliedResBytes, err := applyChange.AppliedResource().AsCompactBytes()
+	if err != nil {
+		return fmt.Errorf("Compact representation of applied resource: %s", err)
+	}
+
+	diff := applyChange.OpsDiff()
```

Comment:
I think this should stay in the diff package, with the rest of the diffing logic and related functions.
Is it not feasible to add this check in the diff package itself?
(Taking a look myself as well)